### ISSUE TITLE
[Feature Request] check out/check Via qr-code/Barcode

### ISSUE THREAD

**Issue by immarkus** [snipe/snipe-it#3152]
#### Expected Behavior (or desired behavior if a feature request)

An option to check out/check in with qr code/Barcode - it will be nice to just enter to a window where you scan all the barcodes (with the smartphone) of the assets You need to checkout/checkin.
for example: u press on "Code bulk check out" and scan one by one.
-----

#### Actual Behavior

There is only one option with the barcode that is directing you to the item's page on snipe-it.

-----

**Comment by VanillaNinjaD** [comment]
Agreed,

bulk check in/out would make the deployment of multiple PCs (a laptop cart for instance) a breeze

**Comment by P4R4NORM4L** [comment]
BUMP

**Comment by rutabagaist** [comment]
I was using the database to bulk checkout items using a barcode scanner until v. 3.7.x Now I'm on 4.2.0 and it no longer works as expected. Previously, on the bulk checkout screen, I could put my cursor into the asset tag field, and scan my barcodes (which are DataMatrix encoded asset tags). I had set my barcode scanner to terminate lines with a CR and this worked very well and allowed for quick checkouts. Each item would scan expected and the field would be ready for the next item with no user intervention. 

Now on version 4.2.0, the bulk checkout will not cleanly accept scanned codes which terminate with CR, TAB, or CRLF. If specify no line ending, the field will locate the asset tag and allow me to click it which then adds it to the checkout list - but this requires a mouse click. In contrast, I can scan assets into the "Audit" field for bulk audits and it works perfectly (just as it did in v 3. I think the helper search feature (which helps find asset tags as one types) is what might be responsible since the bulk audit page does'nt have the same type of field or exhibit this problem.

**Comment by memtech3** [comment]
This is a great idea, many other asset management systems have it.

**Comment by jose-duarte** [comment]
Sees to be similar to #5487 , would be quite a good feature to help out with barcode scanner checkouts

**Comment by tothetopgg** [comment]
+1 for this. I tested it with multiple barcode scanners, but for some reason I have to press enter after every scan or it picks the wrong item. This would help me out a lot. :)

**Comment by aleksandrmetik** [comment]
@uberbrady 
The really important feature, could you please add to the nearest release? Thanks!

**Comment by gogomickey** [comment]
Yes please.  A solution to the bulk checkout helper search box would be great.   It renders a barcode scanner useless for bulk checkouts.

**Comment by zavalalabs** [comment]
BUMP
I would add that this is still important as having a easy way for people to 
Scan an asset in and out of an area or assign multiple items to a user in a simple screen is a must have

**Comment by ReynoldsProductions** [comment]
bumping again. Any word here? Exactly what I am looking to do with a cell phone scan

**Comment by IMA-ChristianPommer** [comment]
That feature would be awesome. We are a research facility with where we want to replace written lend out lists which nobody realy checks. It would be perfect if you could just scan the QR and get to a button to just check it out or in for your account.

**Comment by finn-rm** [comment]
+1

**Comment by alfrede** [comment]
+1 The Feature would be very useful whem you want to inventory testdevices and your Developer wants to Checkout one fast for testing something and give it back and check out

**Comment by jcracine-cimeq** [comment]
+1
This feature would facilitate a more decentralized approach to asset management when the storage facilities are accessible for the users to self-checkout items.

**Comment by Mithondor** [comment]
+1
We're using Snipe-IT at my job now and this would be a big timesaver.
